Sources/Tokenizer.swift

```diff
@@ -71,47 +71,53 @@ public struct SourceMap: Equatable {
   }
 }
 
-public enum Token : Equatable {
+public class Token: Equatable {
+  public enum Kind: Equatable {
+    /// A token representing a piece of text.
+    case text
+    /// A token representing a variable.
+    case variable
+    /// A token representing a comment.
+    case comment
+    /// A token representing a template block.
+    case block
+  }
+  
+  public let contents: String
+  public let kind: Kind
+  public let sourceMap: SourceMap
+  
+  /// Returns the underlying value as an array seperated by spaces
+  public lazy var components: [String] = self.contents.smartSplit()
```

Comment:
Shouldn't that be `private(set)`?

Reply:
jep

Reply:
We may also want to avoid splitting into components if the type is text?

Reply:
We can return just one item with the whole content, but I think this should be handled on a higher level and might as well not happening at all due to the logic flow